src/components/shared/ButtonWithPanel/index.js

```diff
@@ -70,6 +70,20 @@ export class ButtonWithPanel extends React.PureComponent<Props, State> {
     }
   }
 
+  componentDidUpdate(prevProps: Props) {
+    // Handle when the open prop changes
+    const prevOpen = !!prevProps.open;
+    const currentOpen = !!this.props.open;
+    if (prevOpen !== currentOpen) {
+      this.setState({ open: currentOpen });
+      if (currentOpen && !this.state.open) {
+        this.openPanel();
+      } else if (!currentOpen && this.state.open) {
+        this.closePanel();
```

Comment:
No need to handle `closePanel`. Just let it open and don't let it close programmatically. User does it already. Maybe a check like this should be enough `if (prevOpen !== currentOpen && currentOpen)`.